server/src/sas/CompletionProvider.ts

```diff
@@ -338,6 +343,406 @@ export class CompletionProvider {
     });
   }
 
+  public getIndentEdit(
```

Comment:
This file is already large. These functions look cohesive. I'm thinking how about make a separate class say `FormatOnTypeProvider`?

Reply:
OK